src/renderer/vt/invalidate.cpp

```diff
@@ -120,6 +120,12 @@ CATCH_RETURN();
         _circled = circled;
     }
 
+    // If we flushed for any reason other than circling, we don't need to push
+    // the buffer out on EndPaint. We're doing this because we encountered some
+    // VT sequence that required we sync the state of the buffers before passing
+    // it through.

```

Comment:
I also don't quite understand this sentence. The code does make sense to me: "don't flush on end when it's not circling, aka shuffling rows outside of the viewport", but I don't quite get how the comment fits with that... 🤔